Sources/TuistEnvKit/Settings/VersionProvider.swift

```diff
@@ -16,98 +16,63 @@ protocol VersionProviding {
 }
 
 enum VersionProviderError: FatalError {
-    case dataDecodingError(url: Foundation.URL)
-    case responseError(url: Foundation.URL, content: String, statusCode: Int)
+    case noVersionsError
 
     var description: String {
         switch self {
-        case let .dataDecodingError(url):
-            return "Error decoding the response from \(url.absoluteString) as an utf8 string."
-        case let .responseError(url, content, statusCode):
-            return """
-            The request to \(url.absoluteString) return an unsuccessful response with status code \(statusCode) and error body:
-            \(content)
-            """
+        case .noVersionsError:
+            return "Error fetching versions from git."
         }
     }
 
     var type: ErrorType {
         switch self {
-        case .dataDecodingError: return .bug
-        case .responseError: return .bug
+        case .noVersionsError: return .bug
         }
     }
 }
 
 class VersionProvider: VersionProviding {
-    let requestDispatcher: HTTPRequestDispatching
+    let gitHandler: GitHandling
 
-    init(requestDispatcher: HTTPRequestDispatching = HTTPRequestDispatcher()) {
-        self.requestDispatcher = requestDispatcher
+    init(gitHandler: GitHandling = GitHandler()) {
+        self.gitHandler = gitHandler
     }
 
     func versions() -> AnyPublisher<[Version], Error> {
-        requestDispatcher.dispatch(resource: changelogResource())
-            .flatMapLatest { content, _ -> AnyPublisher<[Version], Error> in
-                do {
-                    let versions = try self.parseVersionsFromChangelog(content)
-                    return AnyPublisher(value: versions)
-                } catch {
-                    return AnyPublisher(error: error)
-                }
-            }
-            .eraseToAnyPublisher()
+        do {
+            let content = try gitHandler.lsRemote(url: Constants.gitRepositoryURL)
+            let versions = try parseVersionsFromGit(content)
+            return AnyPublisher(value: versions)
+        } catch {
+            return AnyPublisher(error: error)
+        }
     }
 
     func latestVersion() -> AnyPublisher<Version, Error> {
-        versions().map { versions -> Version in
-            versions.sorted().last!
+        versions().tryMap { versions -> Version in
+            guard let version = versions.last else {
+                throw VersionProviderError.noVersionsError
+            }
+            return version
         }
         .eraseToAnyPublisher()
     }
 
-    func changelogResource() -> HTTPResource<String, Error> {
-        resource(path: "/CHANGELOG.md")
-    }
-
     // MARK: - Fileprivate
 
-    fileprivate func parseVersionsFromChangelog(_ changelog: String) throws -> [Version] {
-        let regex = try NSRegularExpression(pattern: "##\\s+([0-9]+.[0-9]+.[0-9]+)", options: [])
+    fileprivate func parseVersionsFromGit(_ gitOutput: String) throws -> [Version] {
```

Comment:
What do you think about doing the parsing in `GitHandler` itself? Instead of `String`, it could already return a parsed data since there's not a lot of value of just capturing the raw output.

Reply:
I agree this is more a concern of the GitHandler.
We can probably also expose an higher level `remoteTaggedVersions` which returns the tagged versions, instead of `lsRemote`